### PR TITLE
Add WiFiSSLClient (2)

### PR DESCRIPTION
**Pull request by copercini**
Please remove the PR 152 and consider this:

-Moved files out of core folder,
-fixed mbedTLS config file path in platform.txt
-put full bignum.h file insted of include_next (with bug)
-rewrite WiFiSSLClient.cpp and WiFiSSLClient.h  due license problems

### CODE REVIEW

File: platform.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ compiler.warning_flags.all=-Wall -Werror=all -Wextra
...
+compiler.cpreprocessor.flags=-DESP_PLATFORM -DMBEDTLS_CONFIG_FILE="mbedtls/esp_config.h" -DHAVE_CONFIG_H "-I{compiler.sdk.path}/include/config" "-I{compiler.sdk.path}/include/bt" "-I{compiler.sdk.path}/include/driver" "-I{compiler.sdk.path}/include/esp32" "-I{compiler.sdk.path}/include/ethernet" "-I{compiler.sdk.path}/include/fatfs" "-I{compiler.sdk.path}/include/freertos" "-I{compiler.sdk.path}/include/log" "-I{compiler.sdk.path}/include/mdns" "-I{compiler.sdk.path}/include/vfs" "-I{compiler.sdk.path}/include/ulp" "-I{compiler.sdk.path}/include/newlib" "-I{compiler.sdk.path}/include/nvs_flash" "-I{compiler.sdk.path}/include/spi_flash" "-I{compiler.sdk.path}/include/sdmmc" "-I{compiler.sdk.path}/include/openssl" "-I{compiler.sdk.path}/include/app_update" "-I{compiler.sdk.path}/include/tcpip_adapter" "-I{compiler.sdk.path}/include/xtensa-debug-module" "-I{compiler.sdk.path}/include/newlib" "-I{compiler.sdk.path}/include/coap" "-I{compiler.sdk.path}/include/wpa_supplicant" "-I{compiler.sdk.path}/include/expat" "-I{compiler.sdk.path}/include/json" "-I{compiler.sdk.path}/include/mbedtls" "-I{compiler.sdk.path}/include/nghttp" "-I{compiler.sdk.path}/include/lwip"
```
Comment (me-no-dev) [inline platform.txt]: what is different here?

Comment (me-no-dev) [inline platform.txt]: ah I see it

Comment (copercini) [inline platform.txt]: -DMBEDTLS_CONFIG_FILE='"mbedtls/esp_config.h"' to -DMBEDTLS_CONFIG_FILE="mbedtls/esp_config.h"

**Comment by me-no-dev** [comment]
please do not make so many commits ;) test it on your side and commit at once or something :)

**Comment by copercini** [comment]
Sorry, I had to leave the computer and do the shot quickly, only now that I realized some mistakes, really sorry. You can Squash my commits.

I have done now =X

**Comment by me-no-dev** [comment]
Please move all files to separate library and lets keep the same name as ESP8266 (WiFiClientSecure). It will help much in compatibility. Reason for separate library is the fact that Secure client requires extra files that have nothing to do with anything else WiFi and it's better not to be linked to or compiled on the IDE if not used.

**Comment by me-no-dev** [comment]
please pull the latest master so the builds with platformio succeed :)

**Pull request closed without merging**